mailpoet/lib/Automation/Engine/Storage/WorkflowStorage.php

```diff
@@ -211,4 +242,46 @@ private function insertWorkflowVersion(int $workflowId, Workflow $workflow): voi
       throw Exceptions::databaseError($this->wpdb->last_error);
     }
   }
+
+  private function insertWorkflowTriggers(int $workflowId, Workflow $workflow): void {
+    $triggerKeys = [];
+    foreach ($workflow->getSteps() as $step) {
+      if ($step->getType() === Step::TYPE_TRIGGER) {
+        $triggerKeys[] = $step->getKey();
+      }
+    }
+
+    if (!$triggerKeys) {
```

Comment:
Not sure, maybe we should delete existing trigger_keys when no trigger_keys are found. Just to have the correct representation in the database.

To reproduce: If you delete the trigger in your draft and store the draft, the old keys remain in the database.

Reply:
Oh, of course, totally missed that 🤦 Thanks!